File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        n = len(colors)
        ans, mx = 0, neededTime[0]

        for i in range(1, n):
            if colors[i - 1] == colors[i]:
                mn = min(mx, neededTime[i])
                ans += mn
                mx = max(mx, neededTime[i])
            else:
                mx = neededTime[i]

        return ans

File: 1700-minimum-time-to-make-rope-colorful/minimum-time-to-make-rope-colorful.cs
public class Solution {
    public int MinCost(string colors, int[] neededTime) {
        int n = colors.Length;
        int ans = 0, mx = neededTime[0];

        for (int i = 1; i < n; i++) {
            if (colors[i - 1] == colors[i]) {
                ans += Math.Min(mx, neededTime[i]);
                mx = Math.Max(mx, neededTime[i]);
            }
            else {
                mx = neededTime[i];
            }
        }

        return ans;
    }
}